### PR TITLE
sort by name devices and domains

### PR DESCRIPTION
**Pull request by adavida**
it would bother me, it was not sorted by name

I have remedied this problem

david

### CODE REVIEW

**Comment by adavida** [comment]
Hello
I understand that I have to make my sweater resquest be validated
I have always :  Unable to verify (no valid key found) 
thank

File: qui/tray/devices.py [hunk 1]
```diff
@@ -120,7 +120,17 @@ def add_vm(self, _, obj_path):
...
-        self.append(menu_item)
+        
+        # sort function
+        position=0
+        list_tmp={}
+        for i in self.menu_items:
+            list_tmp[str(self.menu_items[i].vm)]=i
+        for i in sorted(list_tmp):
+            if i<str(vm["name"]):
+                position +=1
```
Comment (marmarek) [inline qui/tray/devices.py]: Why not iterate over menu items directly? I.e. without `list_tmp` variable? Something like:
```python
for item_index in self_menu_items:
    if str(self.menu_items[item_index].vm) < str(vm["name"]):
        position += 1
```
Also, code style: please add spaces around operators (`=`, `<` etc).

File: qui/tray/devices.py [hunk 2]
```diff
@@ -215,6 +226,17 @@ def add(self, dev_obj_path: dbus.ObjectPath):
...
+        #sort function
+        list_tmp={}
+        name=dev.backend_domain['name']+':'+dev['ident']
+        for i in self.menu_items:
+            if dev['dev_class'] == i.dev['dev_class']:
+                d_name=str(i.dev.backend_domain['name']+':'+i.dev['ident'])
+                list_tmp[d_name]=1
+        for i in sorted(list_tmp):
+                if i<=str(name):
+                    position +=1
```
Comment (marmarek) [inline qui/tray/devices.py]: Same here.

File: qui/tray/devices.py [hunk 3]
```diff
@@ -120,7 +120,17 @@ def add_vm(self, _, obj_path):
...
+        # sort function
+        position=0
+        list_tmp={}
+        for i in self.menu_items:
+            list_tmp[str(self.menu_items[i].vm)]=i
+        for i in sorted(list_tmp):
+            if i<str(vm["name"]):
+                position +=1
+
+        self.insert(menu_item,position)
```
Comment (marmarek) [inline qui/tray/devices.py]: Space after `,`.

File: qui/tray/domains.py [hunk 4]
```diff
@@ -253,7 +253,14 @@ def update_domain_item(self, _, vm_path):
...
         ])
-        self.tray_menu.add(domain_item)
+
+        position=0
+        for i in self.tray_menu:
+            if i.vm['name']>vm['name']:
+                break
+            position += 1
+            
+        self.tray_menu.insert(domain_item,position)
```
Comment (marmarek) [inline qui/tray/domains.py]: Space after `,`.

**Comment by marmarek** [comment]
> I have always : Unable to verify (no valid key found)

This is because the verification script try to download the key from some keyserver. You can upload it using `gpg --send A21294A857E3C7D2`.

File: qui/tray/devices.py (hunk 1 above)
Comment (adavida) [reply qui/tray/devices.py]: you're right, I corrected
thank

**Comment by adavida** [comment]
here I followed your recommendations
that I have to do to retest the validation

File: qui/tray/devices.py (hunk 1 above)
Comment (marmarek) [reply qui/tray/devices.py]: This still use temporary list (so, traverse the whole list of VMs twice).

**Comment by adavida** [comment]
I have followed your recommendations
and I added the spaces, where I had forgotten them.

File: qui/tray/devices.py (hunk 1 above)
Comment (adavida) [reply qui/tray/devices.py]: I modified, how I valid

File: qui/tray/devices.py [hunk 5]
```diff
@@ -120,7 +120,14 @@ def add_vm(self, _, obj_path):
             self.attached_item = menu_item
 
         self.menu_items[str(obj_path)] = menu_item
-        self.append(menu_item)
+
```
Comment (marmarek) [inline qui/tray/devices.py]: Code style: please remove trailing spaces

File: qui/tray/devices.py [hunk 6]
```diff
@@ -215,7 +222,13 @@ def add(self, dev_obj_path: dbus.ObjectPath):
             return
 
         position = self._position(dev['dev_class'])
-
+
```
Comment (marmarek) [inline qui/tray/devices.py]: Code style: please remove trailing spaces

File: qui/tray/domains.py [hunk 7]
```diff
@@ -253,7 +253,14 @@ def update_domain_item(self, _, vm_path):
...
             "Domain %s is %s" % (vm['name'], str(vm['state']).lower())
         ])
-        self.tray_menu.add(domain_item)
+
+        position = 0
+        for i in self.tray_menu:
+            if i.vm['name'] > vm['name']:
+                break
+            position += 1
+
```
Comment (marmarek) [inline qui/tray/domains.py]: Code style: please remove trailing spaces

Comment (adavida) [reply qui/tray/domains.py]: done
I forgot to post the additional space in my IDE

**Pull request merged**